clmm/theory/parent_class.py

```diff
@@ -238,12 +238,12 @@ def _eval_critical_surface_density(self, z_len, z_src):
 
     def _eval_tangential_shear(self, r_proj, z_cl, z_src):
         delta_sigma = self.eval_excess_surface_density(r_proj, z_cl)
-        sigma_c = self.eval_critical_surface_density(z_cl, z_src)
+        sigma_c = self.cosmo.eval_sigma_crit(z_cl, z_src)
         return delta_sigma/sigma_c
 
     def _eval_convergence(self, r_proj, z_cl, z_src, verbose=False):
         sigma = self.eval_surface_density(r_proj, z_cl, verbose=verbose)
-        sigma_c = self.eval_critical_surface_density(z_cl, z_src)
+        sigma_c = self.cosmo.eval_sigma_crit(z_cl, z_src)
```

Comment:
Is there a specific reason not to call `eval_critical_surface_density`?

Reply:
Hmmm, no not really and I don't recall why I changed it. Maybe because it seems a bit more "native" to call it from the cosmo object than from the modeling object. Actually, we have a lot of redundancy regarding that function, given that `eval_critical_surface_density` and `_eval_critical_surface_density` only call `cosmo.eval_sigma_crit` without adding anything else to it... Maybe we should only keep the one in `Cosmology`?

Reply:
I have no strong opinion on this part, having asked out of curiosity. But I would like to hear what @m-aguena thinks.

Reply:
OK, so after discussing with @m-aguena yesterday, we agreed that these are probably redundant and that we should only keep the one from the Cosmology class. I will open a new issue to deal with it though. So this PR can be merged as is I think.